### PR TITLE
fix(core): adjust createPageAndOpen to prevent setting primary mode repeatedly

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1288

### CODE REVIEW

**Comment by JimmFly** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8069?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8069** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8069?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8083** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8083?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/layouts/workspace-layout.tsx [hunk 1]
```diff
@@ -73,8 +75,27 @@ export const WorkspaceLayoutProviders = ({ children }: PropsWithChildren) => {
...
+  const docsList = docsService.list;
+  const defaultMode =
+    useLiveData(editorSettingService.editorSetting.settings$)
+      .newDocDefaultMode === 'edgeless'
+      ? DocMode.Edgeless
+      : DocMode.Page;
+
+  useEffect(() => {
+    const dispose = currentWorkspace.docCollection.slots.docCreated.on(id => {
+      docsList.setPrimaryMode(id, defaultMode);
```
Comment (forehalo) [inline packages/frontend/core/src/layouts/workspace-layout.tsx]: will it conflict with journal's set?

Comment (JimmFly) [reply packages/frontend/core/src/layouts/workspace-layout.tsx]: I have handled journal in [this line](https://github.com/toeverything/AFFiNE/pull/8069/files#diff-56d4463fdaec0c5b8898f5f1a174d3b22bd2834f3b116565a60e0056bb9cc0fdR39).

Comment (forehalo) [reply packages/frontend/core/src/layouts/workspace-layout.tsx]: move this logic to a separate hook `useAutoDocMode` to reduce component complexity

**Review by EYHN — APPROVED** [review]

**Pull request closed without merging**